src/data/viteTrickFetcher.ts

```diff
@@ -0,0 +1,158 @@
+}
+
+/**
+ * Tricks contain two types of References: 
+ * - recommendedPrerequisites
+ * - variationOf
+ */
+function findUndefinedReferences(allTricks: z.infer<typeof DbTricksTableZod>[]) {
+  
+  let idToTrickLookup: Record<number, (z.infer<typeof DbTricksTableZod>)[]> = createRecordLookup(allTricks);
+
+  // issues define the ID of the Trick and an issue message
+  const issues: [number, string][] = []
+
+  allTricks.forEach( trick => {
+    if(trick.recommendedPrerequisites) {
+      trick.recommendedPrerequisites.forEach( (prerequisitePrimaryKey,i) => {
+        if(prerequisitePrimaryKey[1] !== "official") {
+          issues.push([trick.id, `All recommendedPrerequisites must limit themselves to official tricks (not the case for ${i+1}. Entry)`])
+        }
+        else if(!idToTrickLookup[prerequisitePrimaryKey[0]]) {
+          issues.push([trick.id, `${i+1}. Entry in recommendedPrerequisites references an ID that does not exist (id: ${prerequisitePrimaryKey[0]}).`])
+        }
+        else if(prerequisitePrimaryKey[0] == trick.id) {
+          issues.push([trick.id, `${i+1}. Entry in recommendedPrerequisites references itself (id: ${prerequisitePrimaryKey[0]}).`])
+        }
+
+      })
+    }
+    if(trick.variationOf) {
+      trick.variationOf.forEach( (variationPrimaryKey,i) => {
+        if(variationPrimaryKey[1] !== "official") {
+          issues.push([trick.id, `All variationOf-Entries must limit themselves to official tricks (not the case for ${i+1}. Entry)`])
+        }
+        else if(!idToTrickLookup[variationPrimaryKey[0]]) {
+          issues.push([trick.id, `${i+1}. Entry in variationOf references an ID that does not exist (id: ${variationPrimaryKey[0]}).`])
+        }
+        else if(variationPrimaryKey[0] == trick.id) {
+          issues.push([trick.id, `${i+1}. Entry in variationOf references itself (id: ${variationPrimaryKey[0]}).`])
+        }
+      })
+    }
+  })
+
+  return issues;
+}
+
+
+
+export default async function viteGetAllTricks() {
+  // @ts-expect-error Because there are effectively two TS Projects (Vite Plugin Context 
+  //and Vue Webapp Context), TS gets a bit confused and doesn't think import.meta.url is allowed here
+  const pattern = join(fileURLToPath(import.meta.url), "..", "tricks", "*.yaml");
+  const allFilePaths = await globby(pattern)
+
+  const parsedYamlFiles = await Promise.allSettled(allFilePaths.map( trickFilePath => fetchYamlFile(trickFilePath)))
+  const erroredFiles = parsedYamlFiles.map( (trickYamlFile,i) => trickYamlFile.status === "rejected" ? ({error: trickYamlFile.reason, path: allFilePaths[i]}) : (undefined as never) ).filter(Boolean);
+  const goodFiles = parsedYamlFiles.map( trickYamlFile => trickYamlFile.status === "fulfilled" ? trickYamlFile.value : (undefined as never)).filter(Boolean)
+
+  if(erroredFiles.length > 0) {
+
+    const errorMeta = erroredFiles.map(errorResult => {
+      if(!(errorResult.error instanceof ZodError)) {
+        return errorResult.error+""
```

Comment:
As in the comboFetcher: Why not `String(errorResult.error)`?

Reply:
Laziness, mostly. 
`String(x)` is longer than `x+""` :D 

Could switch to String. It's however important to not use the constructor (`new String(...)`), see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/String#return_value

Reply:
done